evm_arithmetization/src/proof.rs

```diff
@@ -115,65 +115,98 @@ impl PublicValues {
-    /// 256 previous block hashes and current block's hash.
-    pub block_hashes: BlockHashes,
-    /// Extra block data that is specific to the current proof.
-    pub extra_block_data: ExtraBlockData,
+    /// State trie root before the execution of this global state transition.
+    pub state_trie_root_before: H256,
+    /// State trie root after the execution of this global state transition.
+    pub state_trie_root_after: H256,
 }
 
 impl FinalPublicValues {
     /// Extracts final public values from the given public inputs of a proof.
     /// Public values are always the first public inputs added to the circuit,
     /// so we can start extracting at index 0.
     pub fn from_public_inputs<F: RichField>(pis: &[F]) -> Self {
-        assert!(
-            PublicValuesTarget::SIZE - 2 * RegistersDataTarget::SIZE - 2 * MemCapTarget::SIZE
-                <= pis.len()
-        );
+        assert!(FinalPublicValuesTarget::SIZE <= pis.len());
 
         let mut offset = 0;
-        let trie_roots_before =
-            TrieRoots::from_public_inputs(&pis[offset..offset + TrieRootsTarget::SIZE]);
-        offset += TrieRootsTarget::SIZE;
-        let trie_roots_after =
-            TrieRoots::from_public_inputs(&pis[offset..offset + TrieRootsTarget::SIZE]);
-        offset += TrieRootsTarget::SIZE;
-        let block_metadata =
-            BlockMetadata::from_public_inputs(&pis[offset..offset + BlockMetadataTarget::SIZE]);
-        offset += BlockMetadataTarget::SIZE;
-        let block_hashes =
-            BlockHashes::from_public_inputs(&pis[offset..offset + BlockHashesTarget::SIZE]);
-        offset += BlockHashesTarget::SIZE;
-        let extra_block_data =
-            ExtraBlockData::from_public_inputs(&pis[offset..offset + ExtraBlockDataTarget::SIZE]);
+        let state_trie_root_before = get_h256(&pis[offset..offset + TARGET_HASH_SIZE]);
+        offset += TARGET_HASH_SIZE;
+        let state_trie_root_after = get_h256(&pis[offset..offset + TARGET_HASH_SIZE]);
 
         Self {
-            trie_roots_before,
-            trie_roots_after,
-            block_metadata,
-            block_hashes,
-            extra_block_data,
+            state_trie_root_before,
+            state_trie_root_after,
         }
     }
 }
 
 impl From<PublicValues> for FinalPublicValues {
     fn from(value: PublicValues) -> Self {
         Self {
-            trie_roots_before: value.trie_roots_before,
-            trie_roots_after: value.trie_roots_after,
-            block_metadata: value.block_metadata,
-            block_hashes: value.block_hashes,
-            extra_block_data: value.extra_block_data,
+            state_trie_root_before: value.trie_roots_before.state_root,
+            state_trie_root_after: value.trie_roots_after.state_root,
+        }
+    }
+}
+
+/// Memory values which are public once a final block proof is generated.
+/// Note: All the larger integers are encoded with 32-bit limbs in little-endian
```

Comment:
Slightly confusing, since the limb order is LE, but the limb themselves are BE.

Reply:
This is just copy-pasta of identical behaviour for `PublicValues` FWIW